wavefront/templates/tkgi-rolebinding.yaml

```diff
@@ -0,0 +1,15 @@
+{{- if .Values.tkgi.psp.enabled }}
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: rolebinding-default-privileged-sa-ns_wavefront
+  namespace: {{ .Release.Namespace }}
+roleRef:
+  kind: ClusterRole
+  name: psp:pks-privileged
```

Comment:
Should this be just pks-privileged?

Reply:
hmmm, I'm afraid that may not work, there are two _psp_ resources predefined by pks: pks-privileged and pks-restricted, and there is only one _clusterrole_ pre-defined by pks named "psp:restricted" , there is no such _clusterrole_ named "psp:pks-privileged"

Here is my working version: 
1.adding psp policy to the clusterrole wavefront-collector
# Source: wavefront/templates/collector-cluster-role.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  labels:
    app.kubernetes.io/name : wavefront
    helm.sh/chart: wavefront-1.3.2
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io.instance: "wavefront"
    app.kubernetes.io/component: collector
    kubernetes.io/bootstrapping: rbac-defaults
  annotations:
    rbac.authorization.kubernetes.io/autoupdate: "true"
  name: wavefront-collector
rules:
- apiGroups:
  - ""
  resources:
  - events
  - namespaces
  - nodes
  - nodes/stats
  - pods
  - replicationcontrollers
  - services
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - ""
  resources:
  - configmaps
  verbs:
  - get
  - update
  - create
  - list
  - watch
- apiGroups:
  - apps
  resources:
  - daemonsets
  - deployments
  - statefulsets
  - replicasets
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - batch
  resources:
  - jobs
  - cronjobs
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - autoscaling
  resources:
  - horizontalpodautoscalers
  verbs:
  - get
  - list
  - watch
- nonResourceURLs: ["/metrics"]
  verbs:
  - get
**- apiGroups: ['policy']
  resources: ['podsecuritypolicies']
  verbs:     ['use']
  resourceNames:
  - pks-privileged**

2.update the referred clusterrole as psp:restricted(which defined by pks) or just set as wavefront-collector(which already have the authority to use the pks-privileged policy in above step )
# Source: wavefront/templates/project-pacific-rolebinding.yaml
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: rolebinding-default-privileged-sa-ns_wavefront
  namespace: wavefront
**roleRef:
  kind: ClusterRole  
  name: psp:restricted  # or set it as wavefront-collector
  apiGroup: rbac.authorization.k8s.io**
subjects:
  - kind: Group
    apiGroup: rbac.authorization.k8s.io
    name: system:serviceaccounts

Reply:
Thanks for the help. Addressed this part as suggested.